### PR TITLE
SALTO-4798 fetch NS bin instances

### PR DESCRIPTION
**Pull request by dantal4**
Add support in `bin` instances and exclude it by default.

---

_Additional context for reviewer_

---
_Release Notes_: 
Netsuite Adapter:
- Support fetching `bin` instances. To start fetching `bin` instances in your environment, go to your environment configuration, remove `{ name = "bin" }` from the `fetch.exclude.types` section, then add it to the `fetch.include.types` section.

---
_User Notifications_: 
Netsuite Adapter:
- `bin` will be added to `fetch.exclude.types` configuration.

### CODE REVIEW

**Review by omrilit — APPROVED** [review]
Thanks!!

**Comment by omrilit** [comment]
Please wait for a final decision in https://salto-io.atlassian.net/browse/SALTO-4798 before merging this one.

File: packages/netsuite-adapter/src/config.ts [hunk 1]
```diff
@@ -819,6 +822,10 @@ export const LARGE_FOLDERS_EXCLUDED_MESSAGE = 'Some File Cabinet folders exceed
 export const LARGE_TYPES_EXCLUDED_MESSAGE = 'Some types were excluded from the fetch as the elements of that type were too numerous.'
  + ' To include them, increase the types elements\' size limitations and remove their exclusion rules.'
 
+const SUPPORT_BINS_DATE = '2023-10-10'
+export const EXCLUDE_BINS_MESSAGE = `The "Bin" type instances are now supported and will be automatically fetched from ${SUPPORT_BINS_DATE}`
```
Comment (omrilit) [inline packages/netsuite-adapter/src/config.ts:826]: I think it's better to have a PM going over this message

File: packages/netsuite-adapter/src/adapter.ts [hunk 2]
```diff
@@ -415,7 +417,7 @@ export default class NetsuiteAdapter implements AdapterOperations {
       isPartial,
     )
 
-    const updatedConfig = getConfigFromConfigChanges(failures, this.userConfig)
+    const updatedConfig = getConfigFromConfigChanges(failures, this.userConfig, { excludeBins })
```
Comment (omrilit) [inline packages/netsuite-adapter/src/adapter.ts:420]: Let's add a comment with a referenced SALTO ticket to remove the BINs specific code in let's say 1-2 months?
I wouldn't like it to remain in the code in the long run.

**Review by omrilit — APPROVED** [review]
Great work!

File: packages/netsuite-adapter/src/config.ts (hunk 1 above)
Comment (liorn) [reply packages/netsuite-adapter/src/config.ts:826]: "Starting October 10th, 2023, Salto will automatically fetch "Bin" type instances."

Comment (liorn) [reply packages/netsuite-adapter/src/config.ts:826]: Starting October 10th, 2023, "Bin" type instances can be fetched by Salto. To start fetching these instances in your environment, go to your environment configuration and add "{ name = "bin" }" to the types section of fetch.include .

WDYT @dantal4 @omrilit ?

Comment (dantal4) [reply packages/netsuite-adapter/src/config.ts:826]: @liorn they should also remove it from `fetch.exclude.types` in order to fetch it.

Comment (liorn) [reply packages/netsuite-adapter/src/config.ts:826]: Starting October 10th, 2023, "Bin" type instances can be fetched by Salto. 

To start fetching these instances in your environment, go to your environment configuration, remove "{ name = "bin" }" from the fetch.exclude.types section, then add it to the fetch.include.types section.

**Pull request merged**